### ISSUE TITLE
Reparametrize regression model output

### ISSUE THREAD

**Issue by jpata** [jpata/particleflow#351]
Currently, the regression model outputs pt, E, eta, sin_phi, cos_phi.

However, pt and E are highly similar, so the model has to effectively learn almost the same quantity twice, which may hurt the parameter efficiency and loss landscape.
In addition, it means that the model may not necessarily reconstruct particles with a positive mass: `m^2 = E^2 - pt^2 - pz^2`.

It might be useful to reparametrize the loss and the model output in alternative ways, e.g. always assuming a fixed positive mass (e.g. charged pion mass), and then predicting only e.g. pt, inferring energy using the mass relation.

**Comment by erwulff** [comment]
![Screenshot 2024-10-28 at 08 41 52](https://github.com/user-attachments/assets/ab405909-fc11-492b-8439-f08316c15256)

Figure: Regression energy loss, predicting $E$ in red (using current main branch), computing $E$ in green. Train loss in solid lines, validation loss in dotted lines. The model used is the transformer-based MLPF model with FlashAttention.

Predicting $E$ gives lower loss values than assuming a charged pion mass and computing $E$ from $p_t$ and $p_z$. The current main branch is first computing $E'$ assuming a mass of zero and then adding an additional learnable term, $\phi$ to predict the energy as $E = E' + \phi$. This additional learnable term adds flexibility to the energy prediction and seems to result in better performance than simply computing $E$ assuming a charged pion mass via $E^2 = p_t^2 + p_z^2 + m^2$

**Comment by jpata** [comment]
Thanks, given this study, we stick with the current approach of predicting E, rather than assuming a fixed mass value (pion mass) and computing E without any NNs.
